src/ledger/insert_shred.zig

```diff
@@ -462,7 +462,7 @@ pub const ShredInserter = struct {
                 return false;
             }
 
-            if (!shouldInsertCodeShred(&shred, self.max_root.load(.unordered))) {
+            if (!shouldInsertCodeShred(&shred, self.max_root.load(.acquire))) {
```

Comment:
Why use `unordered`? Understand that `unordered` loads have no memory order and the value will be "unspecified" during a data race. This is not a valid solution for this use case.

Here's a basic example that shows the difference: https://godbolt.org/z/YnYxPd8sv. LLVM is allowed to memoize and elide unordered loads.

Reply:
> Why use unordered?

Because it is more efficient. I would use it any time there is no need for a particular ordering of operations within the current thread on the value in question.

> This is not a valid solution for this use case.

How is it invalid? I don't see why we care about a data race here. The atomic is not being used for synchronization. It just needs to get a value that max_root has been set to. It's OK to read a slightly stale value.

Reply:
> Because it is more efficient. 

It's not. There will be no difference at all.

> I would use it any time there is no need for a particular ordering of operations within the current thread on the value in question.

This is what `monotonic` does. Think of `unordered` as telling the compiler that that we don't care about the value read and only care that there is no "technical" data race. It changes the value read from undefined (any non-atomic operation on memory that was atomically operated on is considered UB), to unspecified (where there is nothing that dictates what the value will be). 

> How is it invalid? I don't see why we care about a data race here. The atomic is not being used for synchronization. 

By definition, atomics cannot race. However, we care about the validity of the value under a data-race. Understand that a data-race in this context is not a bug, but a contention between two threads on the same memory object. 

As an absolute rule, `unordered` should never be used in our codebase unless it's very clear why. It's a smell to use, it makes any code around it very suspect, and it is not more efficient in any practical sense.

Reply:
> It's not. There will be no difference at all.

What is the point of having unordered at all, if monotonic always has the same exact performance characteristics? Why would anyone ever use unordered? In another comment you gave examples of optimizations that may be performed by the compiler for unordered which would not be performed for monotonic. Do those optimizations not make the code more efficient?

> > I would use it any time there is no need for a particular ordering of operations within the current thread on the value in question.
> 
> This is what monotonic does. 

Not exactly. Monotonic ordering guarantees that atomic operations on the same value in the same thread are executed in the correct order relative to each other. So there is a minimal amount of ordering that is guaranteed by monotonic.

> As an absolute rule, unordered should never be used in our codebase unless it's very clear why

I'm fine with this. I just wanted to understand the reasoning a bit better, and your godbolt example about eliding the loads out of the loop is satisfying.